src/mitol/google_sheets/api.py

```diff
@@ -0,0 +1,497 @@
+"""API for the Sheets app"""
+import datetime
+import json
+import logging
+import os
+import pickle
+from collections import namedtuple
+from urllib.parse import urljoin
+
+import pygsheets
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db import transaction
+from google.auth.transport.requests import Request  # pylint:disable=no-name-in-module
+from google.oauth2.credentials import Credentials  # pylint:disable=no-name-in-module
+from google.oauth2.service_account import (
+    Credentials as ServiceAccountCredentials,  # pylint:disable=no-name-in-module
+)
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from mitxpro.utils import now_in_utc
+from sheets.constants import (
+    DEFAULT_GOOGLE_EXPIRE_TIMEDELTA,
+    GOOGLE_API_FILE_WATCH_KIND,
+    GOOGLE_API_NOTIFICATION_TYPE,
+    GOOGLE_SERVICE_ACCOUNT_EMAIL_DOMAIN,
+    GOOGLE_TOKEN_URI,
+    REQUIRED_GOOGLE_API_SCOPES,
+    SHEET_RENEWAL_RECORD_LIMIT,
+    SHEET_TYPE_COUPON_ASSIGN,
+    SHEET_TYPE_COUPON_REQUEST,
+    SHEET_TYPE_ENROLL_CHANGE,
+    WORKSHEET_TYPE_REFUND,
+)
+from sheets.models import FileWatchRenewalAttempt, GoogleApiAuth, GoogleFileWatch
+from sheets.utils import (
+    CouponAssignSheetMetadata,
+    CouponRequestSheetMetadata,
+    RefundRequestSheetMetadata,
+    format_datetime_for_google_timestamp,
+    google_timestamp_to_datetime,
+)
```

Comment:
These imports still reference paths from xpro so those should be cleaned up. This file also needs test coverage.